core/include/core/detector.hpp

```diff
@@ -348,7 +350,12 @@ namespace detray
         /** Allowed costructor
          * @param name the detector
          */
-        detector(const std::string &name) : _name(name) {}
+        detector(const std::string &name, vecmem::memory_resource& resource)
```

Comment:
Can this be moved to add_volume_grids()? I think the detector and the grid are built in the csv io, where I also included vecmem in my first draft

Reply:
I explained below about this - I will give it a try if `std::move` works for vector that does not have memory resource because I am not sure as well